src/core/reference/src/op/convert.cpp

```diff
@@ -45,16 +45,6 @@ void jit_convert_vec<float16, float>(jit::Generator& gen, const Xbyak::RegExp& s
     gen.vmovups(gen.yword[dst], f32vec);
 }
 
-template <>
-void jit_convert_vec<float, float16>(jit::Generator& gen, const Xbyak::RegExp& src, const Xbyak::RegExp& dst) {
```

Comment:
This implementation was added for a reason -- to speed up folding I assume. Could you provide additional information on why this implementation was deleted?
Did you measure the impact on load time?

Reply:
When i leave this code, compilation fails since warning as error is set and it's used only inside this file.

We were always doing clamping during f32->f16
https://github.com/openvinotoolkit/openvino/blob/e7c60bb18a6f0e524890cbacc71adc7bb3837b05/src/common/transformations/src/transformations/convert_precision.cpp#L1082
But for some mysterious reason Convert evaluate was left with no clamping. 

I didn't measure load time impact. This should affect only if model is fully in f32 and we intend to infer it in f16. @dmitry-gorokhov do you think this change can significantly impact load time?

Reply:
Currently if there are f32 constants and we want to run it in f16 then this values will be clamped with optimized xbyak.

But if f32 are produced during ConvertPrecision/CF then they will not be clamped. And this changes will affect only such non-trivial cases
![image](https://github.com/openvinotoolkit/openvino/assets/5703039/7ab48c2c-48f5-456e-aeeb-899456f70920)

Reply:
Hi, @pavel-esir : I may missed something, but as mentioned previously this 'during ConvertPrecision/CF then they will not be clamped' maybe expected for some cases, I think that's why we need [ov::float16::quiet_NaN()](https://github.com/pavel-esir/openvino/blob/ca47dcb7397ea2e25240fb24562e01c82f7ee4e6/src/core/include/openvino/core/type/float16.hpp#L210) defined.
and also this changes seems not 'affect only such non-trivial cases', because if we decide to make this changes, plugins spec implement(e.g. [cpu_convert](https://github.com/pavel-esir/openvino/blob/enable_clamp_convert_impl/src/plugins/intel_cpu/src/nodes/common/cpu_convert.cpp)) also need to change to keep align with it.